### PR TITLE
Sec 2

### PR DESCRIPTION
**Pull request by bago2k4**
Update Accountant to use Secretary 2.
As discussed [here](https://github.com/venantius/accountant/issues/2) i finally had a [response](https://github.com/gf3/secretary/issues/77#issuecomment-152267562) from the Secretary guys and i updated the library accordingly.

I have also improved the check on the clicked anchor: it was checking only the path, now i check for the domain to be empty or the same and if the target of the anchor is the same window, if not it shouldn't call `e.preventDefault()`.

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
+(defproject venantius/accountant "0.2-SNAPSHOT"
   :description "Navigation for Single-Page Applications Made Easy."
   :url "http://github.com/venantius/accountant"
   :license {:name "Eclipse Public License"
             :url "http://www.eclipse.org/legal/epl-v10.html"}
-  :dependencies [[org.clojure/clojure "1.7.0"]
-                 [org.clojure/clojurescript "1.7.48"]
-                 [secretary "1.2.3"]
-                 [org.clojure/core.async "0.1.346.0-17112a-alpha"]])
+  :dependencies [[org.clojure/clojure "1.8.0-RC3"]
```
Comment (venantius) [inline project.clj]: Is this required?

**Comment by venantius** [comment]
I'll have to review this in greater depth later this week. My gut says this is fine, but that I probably won't want to merge it into master; rather we'll either create an Accountant 0.2.0 branch or we'll refactor this code so that both Secretary 1 and Secretary 2 can be supported.

**Comment by bago2k4** [comment]
@venantius thanks for the fast reply. Makes sense to me what you are saying since Secretary is doing the same, even if i don't like it, it would be better to create another repo/module IMHO.
Sorry for the delay is opening this.

NB: i didn't mentioned this: if the anchor you click has no href set (checked with empty?) it calls only preventDefault but does nothing. I use this for anchor that have an onClick set and usually call preventDefault to prevent the navigation.

File: project.clj (hunk 1 above)
Comment (bago2k4) [inline project.clj]: nope i just updated it to the latest

**Comment by bago2k4** [comment]
@venantius bump :+1:

**Comment by venantius** [comment]
Sorry, I don't mean to ignore this. Bump is fair. Let me push up a v2 branch; I might ask you to then re-submit this PR against the v2 branch.

**Comment by venantius** [comment]
Okay; there's now a `v2.0.0` branch; could you re-open this PR against that branch? Sorry for the hassle - I do want to get this merged.

**Comment by venantius** [comment]
The motivation for this PR has been somewhat deprecated by @arohner's work generalizing Accountant to work with any routing library. I'm going to close this.

**Pull request closed without merging**